test/iterator.js

```diff
@@ -643,6 +643,99 @@ metatests.testSync('Iterator.chainApply on string', test => {
   test.strictSame(actual, 'h e l l o');
 });
 
+metatests.testSync('Iterator.max no values', test => {
+  const actual = iter([]).max();
+  test.strictSame(actual, undefined);
+});
+
+metatests.testSync('Iterator.max simple', test => {
+  const actual = iter([1, 2, 3, 4, 5]).max();
+  test.strictSame(actual, 5);
+});
+
+metatests.testSync('Iterator.max accessor', test => {
+  const arr = [{ val: 5 }, { val: 4 }, { val: 3 }, { val: 9 }, { val: 4 }];
+  const actual = iter(arr).max(v => v.val);
+  test.strictSame(actual, arr[3]);
+});
+
+metatests.testSync('Iterator.max accessor this', test => {
+  const arr = [5, 4, 3, 2, 1];
+  const actual = iter(arr).max(
+    function(v) {
+      const val = this.curr;
+      this.curr = v;
+      return val;
+    },
+    { curr: 0 }
+  );
+  test.strictSame(actual, 4);
```

Comment:
Why is the result 4 and not 5 here?

Reply:
The order is like this
1. value: 5, curr: 0, accessor returned 0
2. value: 4, curr: 5, accessor returned 5 **our max value**
3. value: 3, curr: 4, accessor returned 4
4. value: 2, curr: 3, accessor returned 3
5. value: 1, curr: 2, accessor returned 2